tests/integration/helpers/cluster.py

```diff
@@ -1476,6 +1476,18 @@ def start(self, destroy_dirs=True):
 
             common_opts = ['--verbose', 'up', '-d']
 
+            images_pull_cmd = self.base_cmd + ['pull']
+            # sometimes dockerhub/proxy can be flaky
+            for i in range(5):
+                try:
+                    run_and_check(images_pull_cmd)
+                    break
+                except Exception as ex:
+                    if i == 4:
+                        raise ex
+                    logging.info("Got exception pulling images: %s", ex)
+                    time.sleep(i * 3)
```

Comment:
Worth using `retry_exception` instead? (and store original exception and throw it, instead of `StopIterator`)